File: cli/python/src/memgo_cli/integrations/agent_detect.py
"""根据已知运行环境识别代理上下文，身份仍由显式参数声明。"""

from __future__ import annotations

import os

_AGENT_CALLER_ENV: tuple[tuple[str, tuple[str, ...]], ...] = (
    ("claude-code", ("CLAUDECODE", "CLAUDE_CODE")),
    ("cursor", ("CURSOR_AGENT", "CURSOR_SESSION_ID")),
    ("codex", ("CODEX_CLI", "OPENAI_CODEX")),
    ("cline", ("CLINE_AGENT", "CLINE")),
    ("continue", ("CONTINUE_AGENT", "CONTINUE_SESSION")),
    ("aider", ("AIDER_SESSION",)),
    ("goose", ("GOOSE_AGENT",)),
    ("windsurf", ("WINDSURF_AGENT",)),
)


def detect_agent_caller() -> str | None:
    """读取已知代理环境变量，返回规范名称或空值。"""
    for name, env_vars in _AGENT_CALLER_ENV:
        if any(os.environ.get(v) for v in env_vars):
            return name
    return None

File: cli/node/src/integrations/agent-detect.ts
/** 检测代理环境，仅触发自动初始化；身份只能由 --agent-caller 显式声明，不能从环境猜测。 */

const AGENT_CALLER_ENV: ReadonlyArray<readonly [string, readonly string[]]> = [
  ['claude-code', ['CLAUDECODE', 'CLAUDE_CODE']],
  ['cursor', ['CURSOR_AGENT', 'CURSOR_SESSION_ID']],
  ['codex', ['CODEX_CLI', 'OPENAI_CODEX']],
  ['cline', ['CLINE_AGENT', 'CLINE']],
  ['continue', ['CONTINUE_AGENT', 'CONTINUE_SESSION']],
  ['aider', ['AIDER_SESSION']],
  ['goose', ['GOOSE_AGENT']],
  ['windsurf', ['WINDSURF_AGENT']],
] as const;

/**
 * 根据已知环境变量识别代理调用方。
 * @returns 识别到的代理名称；未知环境返回空值。
 */
export function detectAgentCaller(): string | null {
  for (const [name, envVars] of AGENT_CALLER_ENV) {
    if (envVars.some((v) => process.env[v])) {
      return name;
    }
  }
  return null;
}
